probably/pgcl/check.py

```diff
@@ -189,6 +197,71 @@ def get_type(program: Program,
     raise Exception("unreachable")
 
 
+def _check_distribution_arguments(prog: Program, expected_type: Type, *parameters: Tuple[Type, Expr]) -> Union[Type, CheckFail]:
+    # this function is copied from the syntax module, which is not pretty, but prevents cyclic imports
+    # also, it works only if constant substitution has been performed
+    # pylint: disable=duplicate-code
+    def _has_variable(expr: Expr) -> bool:
+        if isinstance(expr, UnopExpr) and expr.operator == Unop.IVERSON:
```

Comment:
This line definitely needs a comment for explanation. Why aren't variables within iverson brackets not considered variables?

Reply:
For the linearity checks, there was a reason: kipro2's linear expectations allowed variables both in an Iverson bracket and in the other operand. For example, this is a linear expression: `[x == 2] * y`. That evaluates to `y` if `x == 2` and to `0` otherwise. No need to read this, but here's the paper anyway: https://github.com/moves-rwth/kipro2/blob/master/latticed_k_induction_cav21.pdf

@LKlinke does this even make sense for prodigy? If not, then the two `_has_variable_ functions are actually different.

Reply:
> Why aren't variables within iverson brackets not considered variables?

How do iverson brackets even work in probably? Don't they generate a function? Does this mean we can write something like `[some expression] (x,y,z)` for variables `x,y,z`? The only thing I can find in the code is that the parser doesn't allow iverson brackets in expressions. Then where _can_ they be used?

Reply:
An Iverson bracket `[b]` with a Boolean expression `b` evaluates to `1` if `b` is `true` and `0` otherwise. So `[b] * f` evaluates to `f` if `b` is true and `0` otherwise.

Reply:
Resolved, as this check has been removed